server/src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -130,17 +130,13 @@ public void onMaster() {
         if (logger.isTraceEnabled()) {
             logger.trace("I have been elected master, scheduling a ClusterInfoUpdateJob");
         }
-        try {
-            // Submit a job that will start after DEFAULT_STARTING_INTERVAL, and reschedule itself after running
-            threadPool.schedule(updateFrequency, executorName(), new SubmitReschedulingClusterInfoUpdatedJob());
-            if (clusterService.state().getNodes().getDataNodes().size() > 1) {
-                // Submit an info update job to be run immediately
-                threadPool.executor(executorName()).execute(() -> maybeRefresh());
-            }
-        } catch (EsRejectedExecutionException ex) {
-            if (logger.isDebugEnabled()) {
-                logger.debug("Couldn't schedule cluster info update task - node might be shutting down", ex);
-            }
+
+        // Submit a job that will start after DEFAULT_STARTING_INTERVAL, and reschedule itself after running
+        threadPool.scheduleUnlessShuttingDown(updateFrequency, executorName(), new SubmitReschedulingClusterInfoUpdatedJob());
+
+        if (clusterService.state().getNodes().getDataNodes().size() > 1) {
```

Comment:
This is altering behavior if the thread pool is shutdown. Previously, an `EsRejectedExecutionException` was thrown unconditionally and this part was skipped but now `EsRejectedExecutionException` gets swallowed if the thread pool is shutdown.

So the previous behavior was that `threadPool.executor(executorName()).execute(() -> maybeRefresh());` was skipped (because we went out of the block because `EsRejectedExecutionException` was thrown in the call one line above). The new behavior is that an `EsRejectedExecutionException` is thrown by that line.

Reply:
As discussed, added a try/catch surrounding `threadPool.executor()` in https://github.com/elastic/elasticsearch/pull/37709/commits/822d0c71e20f8959936c2302cb9e0441c8ba9547

Regarding the other concern for posterity noting that: given that `executorName()` is a `MANAGEMENT` ThreadPool i.e. of `ThreadPoolType.SCALING` type (unbounded), it only rejects if shutdown, so we are good here.

Reply:
Correct. Thanks for documenting our discussion result.